### ISSUE TITLE
Incorrect dataflow on clamped value

### ISSUE THREAD

**Issue by joshwatson** [Vector35/binaryninja-api#801]
Given the following MLIL:
![image](https://user-images.githubusercontent.com/5834130/30514446-902a4a94-9ae2-11e7-8f3a-676812607be7.png)

`current_function.medium_level_il.ssa_form[509].get_possible_reg_values('ecx')` should result in a value set of 0-0x20, but it instead results in `<undetermined>`. I've passed the bndb to Brian, and he's pretty sure that it's related the lonely goto.

**Comment by joshwatson** [comment]
Any movement on this one?

**Comment by bpotchik** [comment]
Yes, it's been assigned.

**Comment by joshwatson** [comment]
I'm not sure if this is a change since I filed this issue, but on 1.1.1021-dev, I noticed the following output:
```
>>> current_mlil.ssa_form[446].get_possible_reg_values('ecx')
<const 0x20>
>>> current_mlil.ssa_form[447].get_possible_reg_values('ecx')
<unsigned ranges: [<range: 0x0 to 0x20>]>
>>> current_mlil.ssa_form[511].get_possible_reg_values('ecx')
<undetermined>
```
Where `446` corresponds to `444` in the above image, `447` is the lonely `MLIL_GOTO`, and `510` corresponds to `509` in the image.

I guess it's the `445 @ 00000000` that is messing up dataflow?

**Comment by bpotchik** [comment]
Getting close to a solution for this. I'm experimenting with propagating the branch dependence of a synthetic goto which results in the correct behavior.
<img width="691" alt="screen shot 2018-02-14 at 12 33 04 am" src="https://user-images.githubusercontent.com/29309548/36189174-bf87a3c4-111e-11e8-9828-b1195b307865.png">

**Comment by bpotchik** [comment]
Ended up adding contextual branch sensitivity for dataflow. Fixed in build 1.1.1072-dev.
